Bottom: 29d381d60a5981c0e5eae39ee3c0de328af49274
Top:    fdb07e6e2b121ba4ee83390cd267ea9ed0f4b387
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-08-26 18:55:15 +0100

Introduce global #define for the text cap-height of the PCB font.

This should save the proliferation of rather opaque MIL_TO_COORD(45)'s
in various places.


---

diff --git a/globalconst.h b/globalconst.h
index 1cf29b6..e85b360 100755
--- a/globalconst.h
+++ b/globalconst.h
@@ -126,4 +126,7 @@
 						/* to enable grid drawing */
 	/* size of diamond element mark */
 #define EMARK_SIZE	MIL_TO_COORD (10)
+
+/* (Approximate) capheight size of the default PCB font */
+#define FONT_CAPHEIGHT  MIL_TO_COORD (45)
 #endif
diff --git a/src/change.c b/src/change.c
index 97154fd..5d44355 100644
--- a/src/change.c
+++ b/src/change.c
@@ -839,7 +839,7 @@ static void *
 ChangeTextSize (LayerTypePtr Layer, TextTypePtr Text)
 {
   int value = (Absolute != 0 ? Text->Scale : 0) +
-              (Absolute != 0 ? Absolute : Delta) * 100 / MIL_TO_COORD (45);
+              (Absolute != 0 ? Absolute : Delta) * 100 / FONT_CAPHEIGHT;
 
   if (TEST_FLAG (LOCKFLAG, Text))
     return (NULL);
@@ -915,7 +915,7 @@ static void *
 ChangeElementNameSize (ElementTypePtr Element)
 {
   int value = (Absolute != 0 ? DESCRIPTION_TEXT (Element).Scale : 0) +
-              (Absolute != 0 ? Absolute : Delta) * 100 / MIL_TO_COORD (45);
+              (Absolute != 0 ? Absolute : Delta) * 100 / FONT_CAPHEIGHT;
 
   if (TEST_FLAG (LOCKFLAG, &Element->Name[0]))
     return (NULL);
diff --git a/src/report.c b/src/report.c
index deb6095..f640ef0 100644
--- a/src/report.c
+++ b/src/report.c
@@ -423,7 +423,7 @@ ReportDialog (int argc, char **argv, Coord x, Coord y)
 		 EMPTY (element->Name[0].TextString),
 		 EMPTY (element->Name[1].TextString),
 		 EMPTY (element->Name[2].TextString),
-		 (Coord) (MIL_TO_COORD (45) * element->Name[1].Scale / 100.),
+		 (Coord) (FONT_CAPHEIGHT * element->Name[1].Scale / 100.),
 		 element->Name[1].X, element->Name[1].Y,
 		 TEST_FLAG (HIDENAMEFLAG, element) ? ",\n  but it's hidden" : "",
 		 element->MarkX, element->MarkY,
@@ -464,7 +464,7 @@ ReportDialog (int argc, char **argv, Coord x, Coord y)
 		 "The bounding box is %$mD %$mD.\n"
 		 "%s\n"
 		 "%s", USER_UNITMASK, text->ID, flags_to_string (text->Flags, TEXT_TYPE),
-		 text->X, text->Y, (Coord) (MIL_TO_COORD (45) * (float)text->Scale / 100.),
+		 text->X, text->Y, (Coord) (FONT_CAPHEIGHT * (float)text->Scale / 100.),
 		 text->TextString, text->Direction,
 		 text->BoundingBox.X1, text->BoundingBox.Y1,
 		 text->BoundingBox.X2, text->BoundingBox.Y2,
